packages/components/pages/f-checkout/src/store/checkout.module.js

```diff
@@ -412,39 +423,35 @@ export default {
          * @param {Object} payload - Parameter with the different configurations for the request.
          */
         getGeoLocation: async ({ commit, state }, {
-            url, postData, timeout, tenant
+            url, postData, timeout, tenant, cookies
         }) => {
             let addressCoords;
-
+            const addressDetails = state.address;
+            const shouldLoadAddressFromLocalStorage = addressService.shouldLoadAddressFromLocalStorage(tenant);
             const isAddressInLocalStorage = addressService.isAddressInLocalStorage();
 
-            if (isAddressInLocalStorage) {
-                const storedAddress = addressService.getAddressFromLocalStorage(tenant, false);
-
-                if (addressService.doesAddressInStorageAndFormMatch(storedAddress, state.address)) {
-                    addressCoords = storedAddress.Field1 && storedAddress.Field2 ? [storedAddress.Field2, storedAddress.Field1] : null;
-                    commit(UPDATE_GEO_LOCATION, addressCoords);
-                } else {
-                    commit(UPDATE_GEO_LOCATION, null);
-
-                    const addressDetails = state.address;
-
-                    window.localStorage.setItem('je-full-address-details', JSON.stringify({
-                        PostalCode: addressDetails.postcode,
-                        Line1: addressDetails.line1,
-                        Line2: addressDetails.line2,
-                        Line3: addressDetails.line3,
-                        Line4: addressDetails.line4,
-                        administrativeArea: addressDetails.administrativeArea,
-                        City: addressDetails.locality
-                    }));
+            if (shouldLoadAddressFromLocalStorage) {
+                if (isAddressInLocalStorage) {
+                    const storedAddress = addressService.getAddressFromLocalStorage(tenant, false);
+
+                    addressCoords = getGeoLocationFromAddress(addressDetails, storedAddress);
```

Comment:
Just out of curiosity what is this bit doing in grand scheme of things?

Reply:
Our logic to obtain geolocation is: 1. get it from storage 2. if it doesn't exist in 1, get it from API.
For now we only support restoring geolocation from local storage only. However for some tenants we store such an info in cookies instead. Therefore we need to improve it, so that the geolocation data can be retrieved and use it later on.